### ISSUE TITLE
Get listed in Maven Central

### ISSUE THREAD

**Issue by renepickhardt** [renepickhardt/generalized-language-modeling-toolkit#4]
right now the only way of using the GLMT is to build it with maven. Our goal has to be able to provide various forms:

1.) have a single jar file which can be called with java (which includes all the dependencies)

2.) be able to omit the config.txt by passing arguments via command line

3.) be able to omit the config.txt by using standard values

4.) pushing the software to maven central and being able to call all the classes that we have directly in a persons source code

1 - 3 should be straight forward. for 4.) we might have to update our API and discuss this

**Comment by lschmelzeisen** [comment]
We have a single `.jar` to call and an accompying shell script to execute it.

We are not able to override config with command line arguments, but I do not think this is a problem anymore. since our current config file is vastly differrent from the old one.

There are default values in the config that should suffice most uses.

Pushing to maven is something we might very well want to do in the future, but not the immediate.

**Comment by renepickhardt** [comment]
I agree using is fine. installing and retrieving the toolkit is still a
problem.

On Tue, Jan 6, 2015 at 3:43 PM, Lukas Schmelzeisen <notifications@github.com

> wrote:
> 
> We have a single .jar to call and an accompying shell script to execute
> it.
> 
> We are not able to override config with command line arguments, but I do
> not think this is a problem anymore. since our current config file is
> vastly differrent from the old one.
> 
> There are default values in the config that should suffice most uses.
> 
> Pushing to maven is something we might very well want to do in the future,
> but not the immediate.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/renepickhardt/generalized-language-modeling-toolkit/issues/4#issuecomment-68873140
> .

## 

## 

www.rene-pickhardt.de
http://www.beijing-china-blog.com/

Skype: rene.pickhardt

mobile: +49 (0)157 / 3730 2422    office: +49 (0) 261 / 287 2765    fax:
+49 (0) 267 / 287 100 2765

china mobile: +86 186 2129 5033
